### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by bkircher**
This PR adds support for jsrdbg on Windows. It adds a new directory, `win/`, to the source tree which includes Visual Studio 2012 project files to build jsrdbg as a proper Windows DLL.

Overview:

- Replaced syslog(3) with a file logger on Windows.
- Ported `Thread`, `Mutex`, `Condition` classes and `clock_gettime ` to Win32 API (using `#ifdef _WIN32 ` mostly).
- Added new files `src/tcp_protocol_win32.{h,c}pp` for a Winsock2 implementation of the `TCPProtocol` and `TCPClient` classes.
- Added an iconv(3) replacement for Windows.
- Used default Windows resource loading mechanism to load JS files.
- Public symbols are exported from the DLL using the `JSRDBG_API` macro.
- Added some helper functions to deal with wide character / UTF-16 strings on Windows.
- Updated README.md with instructions for building on Windows.

### CODE REVIEW

File: utils/log.cpp [hunk 1]
```diff
@@ -106,8 +113,94 @@ class UnixLogger : public Logger {
...
+
+    void error( const std::string log, ... ) {
+        va_list args;
+        va_start( args, log );
+        logMessage( ERROR, log, args );
+        va_end( args );
+    }
+
+private:
+    std::ofstream _out;
```
Comment (swojtasiak) [inline utils/log.cpp:170]: Shouldn't we synchronize access to this stream? Loggers are supposed to be thread-safe.

Comment (bkircher) [reply utils/log.cpp:170]: Yes, you're right. This should be synchronized.

**Comment by swojtasiak** [comment]
Looks great! Merging :). We have to release 0.7 in the near future.

**Comment by swojtasiak** [comment]
FileLogger doesn't compile on my gcc 4.9.4. This is probably a bug that was fixed in gcc 5.0. I will take a look at it when I have time.

http://stackoverflow.com/questions/28775673/why-cant-i-move-stdofstrea

**Pull request merged**